### ISSUE TITLE
Build failure: mpv

### ISSUE THREAD

**Issue by ProxyVT** [NixOS/nixpkgs#340574]
### Steps To Reproduce

After I comment `ffmpeg = prev.ffmpeg_7-full;` line from my overlay, system fails to rebuild.

### Build log
<details>

```
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
mesonInstallPhase flags: ''
Installing mpv.1 to /nix/store/grw15gdpi8p1hr19268x53bxvvb41d23-mpv-git-man/share/man/man1
Installing libmpv.so.2.3.0 to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/lib
Installing mpv to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/bin
Installing mpv.desktop to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/share/applications
Installing /build/mpv-5edc897/libmpv/client.h to /nix/store/ahq3j5mlpipj59pfnqhk2vwq7mbgpdwx-mpv-git-dev/include/mpv
Installing /build/mpv-5edc897/libmpv/render.h to /nix/store/ahq3j5mlpipj59pfnqhk2vwq7mbgpdwx-mpv-git-dev/include/mpv
Installing /build/mpv-5edc897/libmpv/render_gl.h to /nix/store/ahq3j5mlpipj59pfnqhk2vwq7mbgpdwx-mpv-git-dev/include/mpv
Installing /build/mpv-5edc897/libmpv/stream_cb.h to /nix/store/ahq3j5mlpipj59pfnqhk2vwq7mbgpdwx-mpv-git-dev/include/mpv
Installing /build/mpv-5edc897/build/meson-private/mpv.pc to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/lib/pkgconfig
Installing /build/mpv-5edc897/etc/mpv.conf to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/share/doc/mpv
Installing /build/mpv-5edc897/etc/input.conf to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/share/doc/mpv
Installing /build/mpv-5edc897/etc/mplayer-input.conf to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/share/doc/mpv
Installing /build/mpv-5edc897/etc/restore-old-bindings.conf to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/share/doc/mpv
Installing /build/mpv-5edc897/etc/mpv.bash-completion to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/share/bash-completion/>
Installing /build/mpv-5edc897/etc/_mpv.zsh to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/share/zsh/site-functions
Installing /build/mpv-5edc897/etc/mpv.metainfo.xml to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/share/metainfo
Installing /build/mpv-5edc897/etc/encoding-profiles.conf to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/etc/mpv
Installing /build/mpv-5edc897/etc/mpv-icon-8bit-16x16.png to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/share/icons/hicolo>
Installing /build/mpv-5edc897/etc/mpv-icon-8bit-32x32.png to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/share/icons/hicolo>
Installing /build/mpv-5edc897/etc/mpv-icon-8bit-64x64.png to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/share/icons/hicolo>
Installing /build/mpv-5edc897/etc/mpv-icon-8bit-128x128.png to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/share/icons/hico>
Installing /build/mpv-5edc897/etc/mpv-gradient.svg to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/share/icons/hicolor/scala>
Installing /build/mpv-5edc897/etc/mpv-symbolic.svg to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/share/icons/hicolor/symbo>
Installing symlink pointing to libmpv.so.2.3.0 to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/lib/libmpv.so.2
Installing symlink pointing to libmpv.so.2 to /nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/lib/libmpv.so
/build/mpv-5edc897/TOOLS /build/mpv-5edc897/build
/build/mpv-5edc897/build
/nix/store/irib0b5fxylczlxdxv6s9pybsf5k3c92-mpv-git/share/applications /build/mpv-5edc897/build
substituteStream() in derivation mpv-git: ERROR: pattern smb\, doesn't match anything in file 'mpv.desktop'
```
</details>

### Additional context

This is my overlay files for mpv package.
<details>
<summary>mpv.nix</summary>

```
{ inputs, outputs, lib, config, pkgs, ... }: {

  imports = [
    ./uosc.nix
  ];

  nixpkgs = {
    overlays = [
      (final: prev: {
        mpv-unwrapped = (prev.mpv-unwrapped.override {
          libplacebo = prev.libplacebo.overrideAttrs (oldAttrs: rec {
            pname = "libplacebo";
            version = "git";
            src = prev.fetchgit {
              url = "https://github.com/haasn/libplacebo.git";
              rev = "efb89342b0c19b9773226624651839686172e88b";
              hash = "sha256-qFD+12dfusY7VVwJDVCtlYjg+duEdeEKWBuTEpBF7O4=";
            };
          });
          #ffmpeg = prev.ffmpeg_7-full;  ######################################### This line!
        }).overrideAttrs ( oldAttrs: rec {
          pname = "mpv";
          version = "git";
          src = prev.fetchgit {
            url = "https://github.com/mpv-player/mpv.git";
            rev = "5edc8973ebdb2f80e1fa5b9c7b0ea7d12fd2fe3b";
            hash = "sha256-VsSB5amtuJ66ljfOJY7XTks2lmmUglOKLKC3qpD3eSw";
          };
          patches = [];
        });
        mpv-git = final.mpv.override {
          scripts = with pkgs; [
            uosc-git
            mpvScripts.thumbfast
          ];
        };
      })
    ];
  };
}

```

</details>

<details>
<summary>uosc.nix</summary>

```
  { inputs, outputs, lib, config, pkgs, ... }: {

  nixpkgs = {
    overlays = [
      (final: prev: {
        uosc-git = prev.mpvScripts.uosc.overrideAttrs (oldAttrs: rec {
          pname = "uosc";
          version = "git";
          src = prev.fetchgit {
            url = "https://github.com/tomasklaen/uosc.git";
            rev = "0f48df315304efd1caf26215526d96660d113dcb";
            hash = "sha256-Ic7XdWqLT7kHdLPRtKGbH26JklMNPBdlTl7sbJSdugc=";
          };
          patches = [];
        });
      })
    ];
  };
}

```

</details>

### Notify maintainers

@AndersonTorres @fpletz @globin @ma27 @tadeokondrak

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.8, NixOS, 24.11 (Vicuna), 24.11.20240907.bf2c184`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.0
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/ulad/.config/nix/nix.conf:/nix/store/wnirp4b1pwbb8096apylpzdk4s88xhjf-cinnamon-settings-daemon-6.2.0/etc/xdg/nix/nix.conf:/nix/store/wnirp4b1pwbb8096apylpzdk4s88xhjf-cinnamon-settings-daemon-6.2.0/etc/xdg/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/ulad/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/ulad/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/ulad/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/dzs6wmlzg8hdh9mjailpghvxrzs43jdn-lix-2.91.0/share`
 - nixpkgs: `/nix/store/4yxsdvhhijjb88hz4vz402g4d8b5pwy3-source`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by AndersonTorres** [comment]
The strange thing to me is that  the error looks unrelated to ffmpeg.

**Comment by fpletz** [comment]
This error is because mpv now detects which protocols are supported and renders the desktop file accordingly. Note that the reporter uses a git snapshot of mpv. This is thus unsupported and has to be fixed in the overlay. It will be fixed with the next version upgrade to 0.39.0: #344452.
